ni_measurement_plugin_sdk_service/_internal/parameter/encoder.py

```diff
@@ -0,0 +1,202 @@
+            message_proto=message_proto, metadata=parameter_metadata, index=i
+        )
+        if parameter_metadata.type == FieldDescriptorProto.TYPE_ENUM:
+            _create_enum_type(
+                file_descriptor=file_descriptor,
+                param=parameter,
+                field_descriptor=field_descriptor,
+            )
+    pool.Add(file_descriptor)
+    return pool.FindMessageTypeByName(f"{file_descriptor.package}.{message_proto.name}")
+
+
+def serialize_default_values(
+    parameter_metadata_dict: Dict[int, ParameterMetadata], service_info: ServiceInfo
+) -> bytes:
+    """Serialize the Default values in the Metadata.
+
+    Args:
+        parameter_metadata_dict (Dict[int, ParameterMetadata]): Configuration metadata.
+
+    Returns:
+        bytes: Serialized byte string containing default values.
+    """
+    default_value_parameter_array = [
+        parameter.default_value for parameter in parameter_metadata_dict.values()
+    ]
+    return serialize_parameters(
+        parameter_metadata_dict, default_value_parameter_array, service_info
+    )
+
+
+def _get_enum_values(param: Any) -> Any:
+    """Get's value of an enum.
+
+    Args:
+        param (Any): A value/parameter of parameter_values.
+
+    Returns:
+        Any: An enum value or a list of enums or the 'param'.
+    """
+    if param == []:
+        return param
+    if isinstance(param, list) and isinstance(param[0], Enum):
+        return [x.value for x in param]
+    elif isinstance(param, Enum):
+        return param.value
+    return param
+
+
+def _create_enum_type(
+    file_descriptor: descriptor_pb2.FileDescriptorProto,
+    param: Any,
+    field_descriptor: FieldDescriptorProto,
+) -> None:
+    """Implement a enum class in 'file_descriptor'.
+
+    Args:
+        file_descriptor (FileDescriptorProto): Descriptor of a proto file.
+
+        param (Any): A value/parameter of parameter_values.
+
+        field_descriptor (FieldDescriptorProto): Descriptor of a field.
+    """
+    if isinstance(param, list):
+        param = param[0]
+    # if there are no enums/param is a different enum and is a python enum, defines a enum field
+    if param.__class__.__name__ not in [
+        enum.name for enum in file_descriptor.enum_type
+    ] and isinstance(param, Enum):
+        enum_descriptor = file_descriptor.enum_type.add()
+        enum_descriptor.name = param.__class__.__name__
+
+        for name, number in param.__class__.__members__.items():
+            enum_value_descriptor = enum_descriptor.value.add()
+            enum_value_descriptor.name = name
+            enum_value_descriptor.number = number.value
+    # checks enum if it's protobuf or python
+    try:
+        field_descriptor.type_name = ProtobufColor.DESCRIPTOR.full_name
+    except TypeError:
+        field_descriptor.type_name = param.__class__.__name__
+
+
+def _create_fields(message_proto: Any, metadata: ParameterMetadata, index: int) -> Any:
```

Comment:
I'd make this `_create_field` since it's only doing one. Also update the parameters in the docstrings - there's some extra ones.